specification/2.0/Specification.adoc

```diff
@@ -3178,22 +3178,23 @@ Metal and dielectric are mixed according to the metalness:
 material = mix(dielectric_brdf, metal_brdf, metallic)
 ----
 
-Taking advantage of the fact that `roughness` is shared between metal and dielectric and that the Schlick Fresnel is used, we can simplify the mix and arrive at the final BRDF for the material:
+The full code is given below:
 
 [source,c]
 ----
-const black = 0
+fresnel_w = (1 - abs(VdotH))^5
 
-c_diff = lerp(baseColor.rgb, black, metallic)
-f0 = lerp(0.04, baseColor.rgb, metallic)
-α = roughness^2
+diffuse_brdf = (1 / π) * baseColor.rgb
+specular_brdf = D(roughness^2) * G(roughness^2) / (4 * abs(VdotN) * abs(LdotN))
 
-F = f0 + (1 - f0) * (1 - abs(VdotH))^5
+dielectric_f0 = 0.04
+dielectric_fresnel = dielectric_f0 + (1 - dielectric_f0) * fresnel_w
+dielectric_brdf = dielectric_fresnel * specular_brdf + (1 - dielectric_fresnel) * diffuse_brdf
```

Comment:
Although mathematically correct, I was hesitating using `mix` for a layering operation because semantically it's not a mix. But that's probably a very weak reason... I also did not use `mix` for the Fresnel itself. 

With `mix` everywhere the code snippet would look like this:

```
fresnel_w = (1 - abs(VdotH))^5

diffuse_brdf = (1 / π) * baseColor.rgb
specular_brdf = D(roughness^2) * G(roughness^2) / (4 * abs(VdotN) * abs(LdotN))

dielectric_f0 = 0.04
dielectric_fresnel = mix(dielectric_f0, 1, fresnel_w)
dielectric_brdf = mix(diffuse_brdf, specular_brdf, dielectric_fresnel)

metal_fresnel = mix(baseColor.rgb, rgb(1, 1, 1), fresnel_w)
metal_brdf = metal_fresnel * specular_brdf

material = mix(dielectric_brdf, metal_brdf, metallic)
```

Do you prefer the fully mixed version, the BRDF mix version or the current?

Reply:
I've no strong opinion here. To me, the mix version seem to be a tad cleaner and easier to comprehend. But maybe others see it differently. I'd be fine with keeping the current version

Reply:
Done, changed to mix everywhere.